### PR TITLE
Remove FunctorEmpty and TraverseEmpty

### PR DESCRIPTION
**Pull request by stephen-lazaro**
Resolves issue #47
cf. https://github.com/typelevel/cats/pull/2405/files
One instance cannot be discovered in the RWSTT tests, currently looking into that... Might need to wait on cats [PR2405](https://github.com/typelevel/cats/pull/2405)?

### CODE REVIEW

File: tests/src/test/scala/cats/mtl/tests/ReaderWriterStateTTests.scala [hunk 1]
```diff
@@ -26,13 +26,13 @@ class ReaderWriterStateTTests extends StateTTestsBase {
...
     }, FunctionK.id[(L, S, ?)]))
-
+  /**
   checkAll("ReaderWriterStateT[Option, Boolean, Int, String, ?]",
     MonadLayerControlTests[ReaderWriterStateT[Option, Boolean, Int, String, ?], Option, (Int, String, ?)]
       .monadLayerControl[Boolean, Boolean])
   checkAll("MonadLayerControl[ReaderWriterStateT[Option, Boolean, Int, String, ?], Option]",
     SerializableTests.serializable(MonadLayerControl[ReaderWriterStateT[Option, Boolean, Int, String, ?], Option]))
-
+  */
```
Comment (stephen-lazaro) [inline tests/src/test/scala/cats/mtl/tests/ReaderWriterStateTTests.scala]: Gotta figure out why this instance is no longer found.

**Comment by LukaJCB** [comment]
Thanks for this :) Looks good so far (though that instance not being found is odd)

File: core/src/main/scala/cats/mtl/instances/all.scala [hunk 2]
```diff
@@ -9,6 +9,5 @@ trait AllInstances extends EitherTInstances
   with RaiseInstances with ReaderTInstances
   with LocalInstances with StateInstances
   with StateTInstances with WriterTInstances
-  with EmptyInstances with ReaderWriterStateTInstances
```
Comment (LukaJCB) [inline core/src/main/scala/cats/mtl/instances/all.scala]: Oh actually it's not odd, you deleted them here :smile:

Comment (stephen-lazaro) [reply core/src/main/scala/cats/mtl/instances/all.scala]: Cool, fixed now!

**Comment by stephen-lazaro** [comment]
Looks like code coverage went down due to the deletion of some hierarchy stuff that referenced `FunctorEmpty`. Is that acceptable or should I wait on PR2405 and then add it back?

**Comment by LukaJCB** [comment]
I don't think we need to add that back, this should be fine :)

**Comment by stephen-lazaro** [comment]
Cool, sounds good to me! Thanks Luka :)

**Pull request merged**